spdx-utils/src/main/kotlin/SpdxUtils.kt

```diff
@@ -37,38 +37,55 @@ val NON_LICENSE_FILENAMES = listOf(
 private fun ByteArray.toHexString(): String = joinToString("") { String.format("%02x", it) }
 
 /**
- * Calculate the [SPDX package verification code][1] for a list of [known SHA1s][sha1sums] of files.
+ * Calculate the [SPDX package verification code][1] for a list of [known SHA1s][sha1sums] of files and [excludes].
  *
  * [1]: https://spdx.org/spdx_specification_2_0_html#h.2p2csry
  */
 @JvmName("calculatePackageVerificationCodeForStrings")
-fun calculatePackageVerificationCode(sha1sums: List<String>): String =
-    sha1sums.sorted().fold(MessageDigest.getInstance("SHA-1")) { digest, sha1sum ->
+fun calculatePackageVerificationCode(sha1sums: List<String>, excludes: List<String> = emptyList()): String {
+    val sha1sum = sha1sums.sorted().fold(MessageDigest.getInstance("SHA-1")) { digest, sha1sum ->
         digest.apply { update(sha1sum.toByteArray()) }
     }.digest().toHexString()
 
+    if (excludes.isEmpty()) {
+        return sha1sum
+    } else {
+        return "$sha1sum (excludes: ${excludes.joinToString()})"
+    }
+}
+
 /**
- * Calculate the [SPDX package verification code][1] for a list of [files].
+ * Calculate the [SPDX package verification code][1] for a list of [files] and paths of [excludes].
  *
  * [1]: https://spdx.org/spdx_specification_2_0_html#h.2p2csry
  */
 @JvmName("calculatePackageVerificationCodeForFiles")
-fun calculatePackageVerificationCode(files: List<File>) =
-    MessageDigest.getInstance("SHA-1").let { digest ->
-        calculatePackageVerificationCode(files.map { file ->
+fun calculatePackageVerificationCode(files: List<File>, excludes: List<String> = emptyList()): String {
+    val digest = MessageDigest.getInstance("SHA-1")
+    return calculatePackageVerificationCode(
+        files.map { file ->
             file.inputStream().use { digest.digest(it.readBytes()).toHexString() }
-        })
-    }
+        },
+        excludes
+    )
+}
 
 /**
  * Calculate the [SPDX package verification code][1] for all files in a [directory]. If [directory] points to a file
  * instead of a directory the verification code for the single file is returned.
+ * All files with the extension ".spdx" are automatically excluded from the generated code.
  *
  * [1]: https://spdx.org/spdx_specification_2_0_html#h.2p2csry
  */
 @JvmName("calculatePackageVerificationCodeForDirectory")
-fun calculatePackageVerificationCode(directory: File) =
-    calculatePackageVerificationCode(directory.walkTopDown().filter { it.isFile }.toList())
+fun calculatePackageVerificationCode(directory: File): String {
+    val (excludes, files) = directory.walkTopDown()
+        .filter { it.isFile }
+        .partition { it.extension == "spdx" }
+    val sortedExcludes = excludes.map { "./${it.relativeTo(directory).invariantSeparatorsPath}" }
+        .sortedWith(compareBy({ it.count { char -> char == '/' } }, { it }))
```

Comment:
Is the sorting really necessary as we walk top-down anyway?

Reply:
Wouldn't then using [walkBottomUp](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/walk-bottom-up.html) do what you want as with it "directories are visited after all their files"?

Reply:
It's still depth-first search, so `./dir/package.spdx` would be visited before `package.spdx`. The difference between `walkBottomUp` and `walkTopDown` is the order directories and their files are visited, but we ignore directories anyway.

Reply:
Ah, right, so you're basically are looking for the functionality in https://github.com/JetBrains/kotlin/pull/2232 to get merged.

Reply:
Yes, I have added a code comment referencing that PR.